clang/include/clang/Sema/Sema.h

```diff
@@ -12970,6 +12970,13 @@ void Sema::addIntelSYCLSingleArgFunctionAttr(Decl *D,
         return;
       }
     }
+    if (CI.getParsedKind() == ParsedAttr::AT_SYCLIntelSchedulerTargetFmaxMhz) {
+      if (ArgInt < 0) {
+        Diag(E->getExprLoc(), diag::err_attribute_requires_positive_integer)
+            << CI.getAttrName() << /*non-negative*/ 1;
+        return;
+      }
+    }
```

Comment:
This should be combined inside addIntelSYCLSingleArgFunctionAttr () function :
if (CI.getParsedKind() == ParsedAttr::AT_SYCLIntelMaxGlobalWorkDim ||
    CI.getParsedKind() == ParsedAttr::AT_SYCLIntelSchedulerTargetFmaxMhz ) {
      if (ArgInt < 0) {
        Diag(E->getExprLoc(), diag::err_attribute_requires_positive_integer)
            << CI.getAttrName() << /*non-negative*/ 1;
        return;
      }

Reply:
`AT_SYCLIntelMaxGlobalWorkDim  `also has an upper-bound check, which is why I kept `AT_SYCLIntelSchedulerTargetFmaxMhz `  in a separate check. There are actually also more FPGA attributes I am adding irefactoring in another patch, which will change this function even more. I will refactor this function then if that sounds ok to you.

Reply:
We definitely need to keep upper bound check for max_global_work_dim kernel attribute (which has a natural limitation of '3'). Several diagnostics and work_gorup_size attributes are dependent on this. @MrSidims could you please comment on this?

Reply:
addIntelSYCLSingleArgFunctionAttr ()  - I added this to handle template parameter support for SYCL FPGA function attributes. So we can add other new function attributes in future here. 

The following attributes below are FPGA memory attributes, i think it would be better if we add  something like addIntelMemoryAttr () instead of adding them inside  addIntelSYCLSingleArgFunctionAttr (). 

IntelFPGAPrivateCopies
IntelFPGABankBits

1 - 1024*1024
IntelFPGABankWidth
IntelFPGANumBanks
IntelFPGAMaxReplicates

@AaronBallman any thought?

Reply:
Thanks for taking a look @smanna12 . I do plan to refactor the functions when I work on the IntelFPGA attributes. You can see https://github.com/intel/llvm/pull/3134. I've uploaded a draft PR (with just one attribute. Others will follow) to get initial feedback from @AaronBallman.